### DISCUSSION TITLE
Feature request: Component variants

### DISCUSSION THREAD

**Discussion by Manueljlin** [uxbox/uxbox#534] [Ideas]
Sorry for spamming all of these 😅 it's the last one for now

This is one of those pretty complex but extremely useful features. It allows an user to keep many variants of the same element under just one component

As an example, see the following screenshot:
![image](https://user-images.githubusercontent.com/11275573/106670572-83357b80-65ad-11eb-8286-d1fea0c31466.png)
These are 12 different states under just one component, even keeping the changed content after switching between them:
![image](https://user-images.githubusercontent.com/11275573/106671683-2aff7900-65af-11eb-8802-7e87831f3452.png)
![image](https://user-images.githubusercontent.com/11275573/106671711-3488e100-65af-11eb-9e6b-f8d3c81180c1.png)
![image](https://user-images.githubusercontent.com/11275573/106671854-6437e900-65af-11eb-8dfe-6f4b67bac158.png)

You folks are just amazing <3 can't wait to see FOSDEM 2021 this weekend.

**Comment by elhombretecla** [comment]
Your feedback is always welcome!!! Dont be shy to make new proposals :)
Working on something similiar to components variants would be great.
Design team will review this and come back with a new components feature with more advanced settings.

**Comment by dialmove** [comment]
I've [proposed a related feature](https://github.com/penpot/penpot/discussions/709) that would allow users to build their own variants for interactive components, using Penpot's existing prototyping capabilities.

**Comment by Crown0815** [comment]
We are currently evaluating Penpot vs. Figma and this feature (present in Figma as [Component Variants](https://help.figma.com/hc/en-us/articles/360056440594-Create-and-use-variants)) was the deciding factor for Figma.
I was quite excited when we came across Penpot since I am a developer who values the usage of SVG and the advantages of OpenSource. However, the missing variants were pointed out by our designer to be a no-go, and I had to agree with them.

Are there any plans of implementing this feature in the near future? According to Taiga, there is a Component 2.0 epic for which this feature would be a great fit.

**Reply by hirunatan** [reply]
Sorry. Components 2.0 is such a big move that we've had to phase it. Variants are not in the first milestone, but they are in the roadmap. You can check the progress here:

https://tree.taiga.io/project/penpot/us/527

**Comment by jcklpe** [comment]
I have previously written a request for this feature: 

- variants: https://github.com/penpot/penpot/issues/768

And this related issue:
- microinteractions: https://github.com/penpot/penpot/issues/769

There's also a forum post that was posted this summer on the subject: 
https://github.com/penpot/penpot/discussions/709

Just posting these links here to try and maximize the network interlinking of these discussions. Penpot should probably think through a little more clearly where things like feature requests, discussions, etc take place, since right now there's at least 3 places they could reside (forums, github discussions, github issues)  Possibly 4 if we count taiga. 

Oh yes there's also a taiga thing thing for this on the roadmap:
- https://tree.taiga.io/project/penpot/us/527

Just noticed that the taiga story is closed and archived. Perhaps @elhombretecla can clarify.

**Comment by SvenPuschmann** [comment]
It is two years later now. Component variants are still not there in PenPot. The Taiga story was told to be closed and archived, but I see it still open. Also, nobody clarified about this. What are the current plans about this feature?
